src/ellipse.rs

```diff
@@ -293,6 +452,58 @@ mod tests {
         assert!((x - y).abs() < 1e-7, "{x} != {y}");
     }
 
+    #[test]
+    fn circular_perimeter() {
+        for radius in [1.0, 0., 1.5, PI, 10.0, -1.0, 1_234_567_890.1] {
+            let circle = Circle::new((0., 0.), radius);
+            let ellipse = Ellipse::new((0., 0.), (radius, radius), 0.);
+
+            let circle_p = circle.perimeter(0.);
+            let ellipse_p = ellipse.perimeter(0.1);
+
+            assert_eq!(
```

Comment:
Asserting equality seems fragile, shouldn't this be an approximate equality test?

Reply:
Maybe. In the Kummer series expansion for the circular case, `x == y`, meaning `h == 0.`. This means the calculation becomes `(x + y) * PI` (`lower == PI` exactly).

`Circle::perimeter` calculates `2.0 * PI * self.radius`. Ellipse effectively calculates `2.0 * x * PI`.

Is this guaranteed to round the same way on all platforms? I'm not sure.

Reply:
I think it is guaranteed to round the same way, because the initial multiply-by-2 should simply increment the float's exponent. I like that the results are identical, but indeed, perhaps it's too fragile.

Reply:
I made the test approximate. Equality indeed is too fragile: even if the calculation itself would be fine, the radii are calculated through the inner affine map, so there are too many moving parts.